### ISSUE TITLE
incompatibility with web worker

### ISSUE THREAD

**Issue by angusmoncrieff** [PepsRyuu/rollup-plugin-hot-css#13]
Hello, and thank you for this plugin!

I'm using nollup, and want to use both rollup-plugin-hot-css and [rollup-plugin-web-worker-loader](https://www.npmjs.com/package/rollup-plugin-web-worker-loader), but they don't seem to be compatible.

When I cause my worker code to execute in my app it fails as follows:

```
Uncaught ReferenceError: window is not defined
    at d486c754-a434-4d9a-a40c-abc6dce4d61f:3
    at d486c754-a434-4d9a-a40c-abc6dce4d61f:28
```

..and clicking through to the code in dev tools:

```javascript
                        ;(function () {
                            window.__css_reload = function () {    // THIS LINE IS HIGHLIGHTED AS CAUSING THE ERROR
                                if (window.__styleLinkTimeout) {
                                    cancelAnimationFrame(window.__styleLinkTimeout);
                                }

                                window.__styleLinkTimeout = requestAnimationFrame(() => {
                                    var link = document.querySelector('link[href*="assets/styles-bebce691.css"]');

                                    if (link) {
                                        if (!window.__styleLinkHref) {
                                            window.__styleLinkHref = link.getAttribute('href');
                                        }

                                        var newLink = document.createElement('link');
                                        newLink.setAttribute('rel', 'stylesheet');
                                        newLink.setAttribute('type', 'text/css');
                                        newLink.setAttribute('href', window.__styleLinkHref + '?' + Date.now());
                                        newLink.onload = () => {
                                            link.remove();
                                        };

                                        document.head.appendChild(newLink);
                                    }
                                });
                            }
                        })();
                    (function () {
	'use strict';

	// This worker provides the timing service for scales being played and for the metronome.  // MY CODE! 👇
	// From https://github.com/cwilso/metronome
	// console.log('worker script');
	var timerID = null;
	var interval = 100;

	function tick() {
	  // console.log('metronomeWorker tick');
	  postMessage("tick");
	}

	onmessage = function onmessage(e) {
	  if (e.data === "start") {
	    timerID = setInterval(tick, interval); // console.log("metronomeWorker - START - timerID, interval: ", timerID, interval);
	  } else if (e.data.interval) {
	    interval = e.data.interval; // console.log("metronomeWorker - SET INTERVAL - timerID, new interval: ", timerID, interval);

	    if (timerID) {
	      clearInterval(timerID); // timerID = setInterval(function () { console.log('metronomeWorker tick (b)'); postMessage("tick"); }, interval)

	      timerID = setInterval(tick, interval);
	    }
	  } else if (e.data === "stop") {
	    // console.log("metronomeWorker: STOP - timerID: ", timerID);
	    clearInterval(timerID);
	    timerID = null;
	  } else {
	    console.log('received message: ', e.data);
	    postMessage('test-from-worker');
	  }
	};

}());
```

..all the stuff at the bottom is my worker.

So at a guess it looks like the hot-css plugin is injecting code at the top, but it depends on 'window' existing which it doesn't in a web worker. Indeed, AFAIK css is not really a thing at all within a web worker.

Is there any configuration I can do to avoid this error?

Any help or insight very much appreciated!

**Comment by PepsRyuu** [comment]
It's a bug, I'll need to add a check into the hot-css plugin so that it only adds the code to the original entry module. Will have a look into this.

**Comment by PepsRyuu** [comment]
So the way that that web worker plugin works makes it impossible to determine if it's the main entry point or not. So I added a window check and that seems to do the trick. Not ideal, but it works! Released in ```0.5.1```.

**Comment by angusmoncrieff** [comment]
Wow that was fast!

Works for me - thank you!
